src/Swift.Bindings/src/Emitter/StringCSharpEmitter/Handler/MethodHandler.cs

```diff
@@ -325,7 +347,14 @@ public void HandleReturnType()
         {
             if (!MarshallingHelpers.MethodRequiresIndirectResult(MethodDecl, ParentDecl, TypeDatabase))
             {
-                var returnType = MethodDecl.CSSignature.First().CSTypeIdentifier.Name;
+                var argument = MethodDecl.CSSignature.First();
+                var returnType = argument.CSTypeIdentifier.Name;
+                var typeRecord = MarshallingHelpers.GetType(argument, TypeDatabase.Registrar);
```

Comment:
I feel that we now operate on two abstraction layers in this call. We have high-level "MethodRequiresIndirectResult" which calls type database and low level `MarshallingHelpers.GetType` which can lead to confusing behaviour as discussed in previous comment.